Blog post about the new release (#973)
datenanfragen/website: content/de/blog/v2-release/index.md

{

>     "title": "Datenanfragen.de v2: Nagelneue Oberfläche und Beschwerdegenerator",

Reviewer: "v2" feels more English to me. I'd say, in German you'd more commonly say "2.0".

```suggestion
    "title": "Datenanfragen.de 2.0: Nagelneue Oberfläche und Beschwerdegenerator",
```

    "type": "blog",

>     "date": "2022-12-07T12:00:00+02:00",

Reviewer: The slug should be translated accordingly.

Author: What do you mean?

Reviewer: Use something other than `v2-release` for the slug.

Author: Why?

Author: But this is perfectly fine in both languages?

    "description": "Gestern haben wir die neue Version der Webseite mit vielen neuen Features veröffentlicht: Eine neue Oberfläche für den Anfragengenerator, eine umfangreichere Anfragenhistorie, eine schicke neue Startseite und ein Generator für Beschwerden und Mahnungen.",

>     "featured_image": "stockings-feature",

Reviewer: The background needs to be a lot darker and bluer for consistency and color contrast.

>     "tags": [ "Beschwerdegenerator", "Neue Nutzer_innenoberfläche", "Update", "Open Source" ],

Reviewer: ```suggestion
    "tags": [ "Beschwerdegenerator", "Neue Oberfläche", "Update", "Open Source" ],
```

    "authors": [ "zner0L" ]
}

Gestern haben wir Euch etwas ganz Besonderes in den Nikolausstiefel getan: Das neue Release von unserer Webseite mit vielen tollen neuen Features! Wir haben in den letzten Monaten viel Zeit damit verbracht, die Nutzer_innenoberfläche zu updaten und haben uns viel Mühe gegeben es so einfach und barrierearm wie möglich zu machen, Anfragen an Unternehmen und andere Datenverantwortliche zu schicken. Dank des [Prototype Fund](https://prototypefund.de/), der von der [Open Knowledge Foundation Deutschland](https://okfn.de/) betreut und vom [Bundesministerium für Bildung und Forschung](https://bmbf.de) finanziert wird, und Lorenz und Benni gefördert hat, konnten wir mehr Zeit als üblich in dieses Release stecken (mehr dazu auf unserer {{< link slug="verein/transparency#12-externe-f%c3%b6rdermittel-die-mitglieder-erhalten-haben" text="Transparenzseite" >}}). Jetzt sind wir sehr aufgeregt und stolz, Euch alle neuen Features vorstellen zu können. Wenn Du einen Blick auf den Code werfen möchtest, kannst Du [alle Änderungen auf GitHub finden](https://github.com/datenanfragen/website/pull/921).

> 

Reviewer: Any reason you're not including the featured image here like in all other posts (also for EN).

> Gestern haben wir Euch etwas ganz besonderes in den Nikolausstiefel getan: Das neue Release von unserer Webseite mit vielen tollen neuen Features! Wir haben in den letzten paar Monaten viel Zeit damit verbracht, die Nutzer_innenoberfläche zu updaten und haben uns viel Mühe gegeben es so einfach und barrierearm wir möglich zu machen, Anfragen an Unternehmen und anderen Datenverantwortliche zu schicken. Dank des [Prototype Fund](https://prototypefund.de/), der von der [Open Knowledge Foundation Deutschland](https://okfn.de/) betreut und vom [Bundesministerium für Bildung und Forschung](https://bmbf.de) finanziert wird und Lorenz und Benni gefördert hat, konnten wir mehr Zeit als üblich in dieses Release stecken (mehr dazu auf unserer {{< link slug="verein/transparency#12-externe-f%c3%b6rdermittel-die-mitglieder-erhalten-haben" text="Transparenzseite" >}}). Jetzt sind wir sehr aufgeregt und stolz Euch alle neuen Features vorstellen zu können. Wenn Du einen Blick auf den Code werfen möchtest, kannst Du [alle Änderungen auf GitHub finden](https://github.com/datenanfragen/website/pull/921).

Reviewer: ```suggestion
Gestern haben wir Euch etwas ganz Besonderes in den Nikolausstiefel getan: Das neue Release von unserer Webseite mit vielen tollen neuen Features! Wir haben in den letzten Monaten viel Zeit damit verbracht, die Nutzer_innenoberfläche zu updaten und haben uns viel Mühe gegeben es so einfach und barrierearm wie möglich zu machen, Anfragen an Unternehmen und andere Datenverantwortliche zu schicken. Dank des [Prototype Fund](https://prototypefund.de/), der von der [Open Knowledge Foundation Deutschland](https://okfn.de/) betreut und vom [Bundesministerium für Bildung und Forschung](https://bmbf.de) finanziert wird, und Lorenz und Benni gefördert hat, konnten wir mehr Zeit als üblich in dieses Release stecken (mehr dazu auf unserer {{< link slug="verein/transparency#12-externe-f%c3%b6rdermittel-die-mitglieder-erhalten-haben" text="Transparenzseite" >}}). Jetzt sind wir sehr aufgeregt und stolz, Euch alle neuen Features vorstellen zu können. Wenn Du einen Blick auf den Code werfen möchtest, kannst Du [alle Änderungen auf GitHub finden](https://github.com/datenanfragen/website/pull/921).
```

## Mehrseitiger Anfragengenerator

> Wir haben den Generatorablauf komplett neu gestaltet, um ihn aufzuräumen und das Senden vieler Anfragen an viele verschiedene Unternehmen auf einen Schlag zu erleichtern. Während der alte Generator, den wir weiterhin unter {{< link slug="/g" text="/g" >}} zur Verfügung stellen für alle die lieber die alte Oberfläche benutzen wollen, mit einer Seite alles erledigen will, haben wir im neuen Generator den Anfrageprozess auf mehrere Seiten aufgeteilt und begleiten Dich jetzt bis zum Versenden der Anfrage. In der Unternehmensauswahlphase schlagen wir nun außerdem Unternehmenspacks vor, sodass neuen Nutzer_innen der Einstieg erleichtert wird. Im nächsten Schritt kannst Du nun auch auswählen, welche Untereinheiten eines Verantwortlichen Du in Deine Anfrage aufnehmen willst und festlegen, ob es sich um eine Anfrage in Bezug auf Tracking handelt, bei der Du noch nicht weist wie die Verantwortliche dich identifizieren kann.

Reviewer: ```suggestion
Wir haben den Generatorablauf komplett neu gestaltet, um ihn aufzuräumen und das Senden vieler Anfragen an viele verschiedene Unternehmen auf einen Schlag zu erleichtern. Während der alte Generator, den wir weiterhin unter {{< link slug="/g" text="/g" >}} zur Verfügung stellen für alle, die lieber die alte Oberfläche benutzen wollen, mit einer Seite alles erledigen will, haben wir den Anfrageprozess im neuen Generator auf mehrere Seiten aufgeteilt und begleiten Dich jetzt bis zum Versenden der Anfrage. In der Unternehmensauswahlphase schlagen wir nun außerdem Pakete an Unternehmen vor, sodass neuen Nutzer_innen der Einstieg erleichtert wird. Im nächsten Schritt kannst Du nun auch auswählen, welche Untereinheiten eines Verantwortlichen Du in Deine Anfrage aufnehmen willst und festlegen, ob es sich um eine Anfrage in Bezug auf Tracking handelt, bei der Du noch nicht weißt, wie die Verantwortliche dich identifizieren kann.
```

> {{< img name="multi-page-generator-de" alt="Screenshot von drei Seiten des Anfragegenrators (von links nach rechts): Das erste Bild zeigt die Auswahl des Anfragetyps mit den Optionen “Auskunft über meine Daten erhalten.”, “(Teile von) Daten über mich löschen.”, “Daten über mich korrigieren.” and “Widerspruch gegen Direktwerbung einlegen”. Das zweite Bild zeigt die Unternehmensauswahlseite mit einer Suchleiste oben und darunter die Unternehmenspacks mit den Titeln: “Adresshandel und -management”, “Auskunfteien”, “Handel” and “Unterhaltung”. Das dritte Bild zeigt den Identifikationsdatendialog mit einigen Eingabefeldern die nach Identifikationsdaten für den “Datenanfragen.de e. V.“ frageb wie etwa Name, E-Mail-Adresse and Adresse. Unter den Eingabefeldern ist ein Knopf um die Anfrage abzusenden." caption="Drei Seiten des neuen Generators: Die Auswahl des Anfragentyps, die Unternehmenssuche und die Ansicht zum Ausfüllen von Identifikationsdaten." >}}

Reviewer: The third screenshot is in English.

> {{< img name="multi-page-generator-de" alt="Screenshot von drei Seiten des Anfragegenrators (von links nach rechts): Das erste Bild zeigt die Auswahl des Anfragetyps mit den Optionen “Auskunft über meine Daten erhalten.”, “(Teile von) Daten über mich löschen.”, “Daten über mich korrigieren.” and “Widerspruch gegen Direktwerbung einlegen”. Das zweite Bild zeigt die Unternehmensauswahlseite mit einer Suchleiste oben und darunter die Unternehmenspacks mit den Titeln: “Adresshandel und -management”, “Auskunfteien”, “Handel” and “Unterhaltung”. Das dritte Bild zeigt den Identifikationsdatendialog mit einigen Eingabefeldern die nach Identifikationsdaten für den “Datenanfragen.de e. V.“ frageb wie etwa Name, E-Mail-Adresse and Adresse. Unter den Eingabefeldern ist ein Knopf um die Anfrage abzusenden." caption="Drei Seiten des neuen Generators: Die Auswahl des Anfragentyps, die Unternehmenssuche und die Ansicht zum Ausfüllen von Identifikationsdaten." >}}

Reviewer: ```suggestion
{{< img name="multi-page-generator-de" alt="Screenshot von drei Seiten des Anfragengenerators (von links nach rechts): Das erste Bild zeigt die Auswahl des Anfragetyps mit den Optionen „Auskunft über meine Daten erhalten.“, „(Teile von) Daten über mich löschen.“, „Daten über mich korrigieren.“ and „Widerspruch gegen Direktwerbung einlegen“. Das zweite Bild zeigt die Unternehmensauswahlseite mit einer Suchleiste oben und darunter die Unternehmenspacks mit den Titeln: „Adresshandel und -management“, „Auskunfteien“, „Handel“ and „Unterhaltung“. Das dritte Bild zeigt den Identifikationsdatendialog mit einigen Eingabefeldern, die nach Identifikationsdaten für den „Datenanfragen.de e. V.“ fragen, wie etwa Name, E-Mail-Adresse und Adresse. Unter den Eingabefeldern ist ein Knopf, um die Anfrage abzusenden." caption="Drei Seiten des neuen Generators: Die Auswahl des Anfragentyps, die Unternehmenssuche und die Ansicht zum Ausfüllen von Identifikationsdaten." >}}
```

## Umfangreiche Anfragenhistorie

> Wenn Du unsere Webseite regelmäßig nutzt kennst Du vermutlich unsere “Meine Anfragen” Seite. Wenn Du es nicht in den {{< link slug="/privacy-controls" text="Datenschutzeinstellungen" >}} ausgeschaltet hast, speichern wir alle Anfragen, die Du verschickst in Deinem Browser und geben Dir darüber eine Übersicht. Bisher war das nur eine unübersichtliche Tabelle, die wir schon seit Jahren verbessern und als richtiges Anfragenverwaltungstool aufbauen wollen. In unserem neuen Release erstellen wir jetzt einen Vorgang für jede Deine Anfragen, der alle Nachrichten enthält, die etwas mit ihr zu tun haben einschließlich Mahnungen und Beschwerden. Du kannst Informationen über Nachrichten, die Du von der Verantwortlichen erhälst, hinzufügen, sodass du alles an einem Ort gesammelt hast, falls Du dann eine Beschwerde senden möchtest. Wir speichern den Status all Deiner Anfragen, damit Du ihren aktuellen Stand verfolgen kannst, und erinnern Dich mit einer Badge auf allen Seiten daran, falls es überfällige Anfragen gibt für die Du eine Mahnung verschicken solltest.

Reviewer: ```suggestion
Wenn Du unsere Webseite regelmäßig nutzt, kennst Du vermutlich unsere „Meine Anfragen“-Seite. Wenn Du es nicht in den {{< link slug="/privacy-controls" text="Datenschutzeinstellungen" >}} ausgeschaltet hast, speichern wir alle Anfragen, die Du verschickst, in Deinem Browser und geben Dir darüber eine Übersicht. Bisher war das nur eine unübersichtliche Tabelle, die wir schon seit Jahren verbessern und als richtiges Anfragenverwaltungstool ausbauen wollten. In unserem neuen Release erstellen wir jetzt einen Vorgang für jede Deine Anfragen, der alle Nachrichten enthält, die etwas mit ihr zu tun haben, einschließlich Mahnungen und Beschwerden. Du kannst Informationen über Nachrichten, die Du von der Verantwortlichen erhältst, hinzufügen, sodass Du alles an einem Ort gesammelt hast, falls Du dann eine Beschwerde senden möchtest. Wir speichern den Status all Deiner Anfragen, damit Du ihren aktuellen Stand verfolgen kannst, und erinnern Dich auf allen Seiten daran, falls es überfällige Anfragen gibt, für die Du eine Mahnung verschicken solltest.
```

> {{< img name="my-requests" alt="Screenshot der “Meine Anfragen” Seite. Unter dem Erklärungstext stehen mehrere Vorgänge mit verschiedenen Status, wie etwa “wartet auf Antwort”, “überfällig” and “abgeschlossen”. Ein Vorgang ist aufgeklappt und enthält zwei Nachrichten, eine von der Nutzer_in und eine des angefragten Unternehmens." caption="Die neue Anfragenhistorie auf der “Meine Anfragen“ Seite." >}}

Reviewer: ```suggestion
{{< img name="my-requests" alt="Screenshot der „Meine Anfragen“-Seite. Unter dem Erklärungstext stehen mehrere Vorgänge mit verschiedenen Status, wie etwa „wartet auf Antwort“, „überfällig“ und „abgeschlossen“. Ein Vorgang ist aufgeklappt und enthält zwei Nachrichten, eine von der Nutzer_in und eine des angefragten Unternehmens." caption="Die neue Anfragenhistorie auf der „Meine Anfragen“-Seite." >}}
```

## Beschwerdegenerator und Anfragereaktionen

> Die neuen “Meine Anfragen” Seite kommt erst mit einem weiteren Feature so richtig zur Geltung, auf das wir besonders stolz sind: Basierend auf Deiner Anfragenhistorie geben wir Dir die Möglichkeit zu “reagieren”. Wenn Du Dich dazu entscheidest, helfen wir Dir einzuschätzen, welche Fehler eine Verantwortliche möglicherweise gemacht hat, indem wir Dir ein paar Fragen stellen. Daraus generieren wir dann eine Mahnung oder sogar eine richtige Beschwerde. Auch wenn wir nochmal deutlich machen wollen, dass wir keine Anwälte sind und keine Rechtsberatung für spezifische Fälle anbieten können, hoffen wir trotzdem, dass diese Feature es Dir leichter macht, Dich bei den Aufsichtsbehörden zu beschwerden. Du bist mit Deiner Überforderung von Unternehmensantworten nicht mehr allein.

Reviewer: ```suggestion
Die neuen „Meine Anfragen“-Seite kommt erst mit einem weiteren Feature so richtig zur Geltung, auf das wir besonders stolz sind: Basierend auf Deiner Anfragenhistorie geben wir Dir die Möglichkeit zu „reagieren“. Wenn Du Dich dazu entscheidest, helfen wir Dir einzuschätzen, welche Fehler eine Verantwortliche möglicherweise gemacht hat, indem wir Dir ein paar Fragen stellen. Daraus generieren wir dann eine Mahnung oder sogar eine richtige Beschwerde. Auch wenn wir nochmal deutlich machen wollen, dass wir keine Anwälte sind und keine Rechtsberatung für spezifische Fälle anbieten können, hoffen wir trotzdem, dass diese Feature es Dir leichter macht, Dich bei den Aufsichtsbehörden zu beschweren. Du bist mit Deiner Überforderung von Unternehmensantworten nicht mehr allein.
```

Reviewer: I already didn't particularly like the English version but I doubt many people are going to understand "Du bist mit Deiner Überforderung von Unternehmensantworten nicht mehr allein.".

> {{< img name="complaint-generator" alt="Screenshot des Mahnungsgenerators. Ganz oben steht die Frage “Gab es ein Problem mit Deiner Anfrage?” und darunter verschiedene Buttons für verschiedene Arten von Problemen. Ein Beispiel für einen Buttontext: “Das Unternehmen fragt nach zusätzlichen Daten zur Authentifizierung.”" caption="Erste Seite des Mahnungsgenerators." >}}

Reviewer: ```suggestion
{{< img name="complaint-generator" alt="Screenshot des Mahnungsgenerators. Ganz oben steht die Frage „Gab es ein Problem mit Deiner Anfrage?“ und darunter verschiedene Buttons für verschiedene Arten von Problemen. Ein Beispiel für einen Buttontext: „Das Unternehmen fragt nach zusätzlichen Daten zur Authentifizierung.“" caption="Erste Seite des Mahnungsgenerators." >}}
```

## Schicke Startseite

> Eine weitere Sache, die schon seit Jahren auf unserer Wunschliste steht ist die Verbesserung underer Startseite, die nie wirklich gut zu dem gepasst hat, was wir eigentlich machen. Jetzt leiten wir Nutzer_innen zum Anfragengenerator mit klaren Calls-to-action und geben einen kurzen Überblick wie es funktioniert. Selbst für erfahrene Nutzer_innen lohnt sich hin und wieder der Besuch der Startseite: Du kannst dort dank unseres Erinnerungswidgets direkt auf überfällige Anfragen reagieren und außerdem finden zeigen wir dort unsere neusten Blogartikel.

Reviewer: ```suggestion
Eine weitere Sache, die schon seit Jahren auf unserer Wunschliste steht, ist die Verbesserung unserer Startseite, die nie wirklich gut zu dem gepasst hat, was wir eigentlich machen. Jetzt leiten wir Nutzer_innen zum Anfragengenerator mit klaren Calls-to-action und geben einen kurzen Überblick, wie der Prozess funktioniert. Selbst für erfahrene Nutzer_innen lohnt sich hin und wieder der Besuch der Startseite: Du kannst dort dank unseres Erinnerungswidgets direkt auf überfällige Anfragen reagieren und außerdem zeigen wir dort unsere neusten Blogartikel.
```

> {{< img name="homepage" alt="Screenshot der Startseite. Sie hat eine Kopfleiste mit dem Menü und einer Badge, die auf die überfälligen Anfragen hinweist (14) und darunter ein Widget mit dem Text: “Du hast 14 überfällige Anfragen. Reagiere jetzt oder markiere sie als zufriedenstellend.”. Darunter steht groß der Satz: “Du hast ein Recht auf Vergessenwerden” und daneben sind vier Buttons für die Anfragetypen." caption="Unsere neue Startseite mit Erinnerungswidget." >}}

Reviewer: ```suggestion
{{< img name="homepage" alt="Screenshot der Startseite. Sie hat eine Kopfleiste mit dem Menü und einer Badge, die auf die überfälligen Anfragen hinweist (14) und darunter ein Widget mit dem Text: „Du hast 14 überfällige Anfragen. Reagiere jetzt oder markiere sie als zufriedenstellend.“. Darunter steht groß der Satz: „Du hast ein Recht auf Vergessenwerden.“ und daneben sind vier Buttons für die Anfragetypen." caption="Unsere neue Startseite mit Erinnerungswidget." >}}
```

## Refactoring: TypeScript, zustand und weitere Verbesserungen

> Im Maschinenraum (im Code) haben wir auch viel verändert. Manche der Änderungen haben wir auch schon im laufe der letzten Monate veröffentlicht. Wir sind zu [TypeScript](https://www.typescriptlang.org/) gewechselt um Fehler früher zu entdecken (und Mädchen haben wir viele gefunden) und nutzen jetzt [zustand](https://github.com/pmndrs/zustand) für Statemanagement, in der Hoffnung, dass das Race condition bugs im Code behebt. Das heißt auch, dass wir die alten gespeicherten Anfragen in eine neue Speicherart überführt haben. Wir hoffen, dass alles gut gegangen ist, aber falls Du irgendwelche Probleme bemerkt hast und/oder irgendwelche Anfragen vermisst, [schreib uns eine E-Mail](mailto:dev@datenanfragen.de). Außerdem haben wir den Code ordentlich aufgeräumt und seine alten und disfunktionalen Ecken entfernt. Und wir veröffentlichen jetzt [ein Package unserer Generators](https://www.npmjs.com/package/@datenanfragen/components), dass Du in Deinen eigenen Code einbauen kannst, z.B. wenn du es in Deine Webseite einbetten willst.

Reviewer: ```suggestion
Im Maschinenraum (im Code) haben wir auch viel verändert. Manche der Änderungen haben wir auch schon im Laufe der letzten Monate veröffentlicht. Wir sind zu [TypeScript](https://www.typescriptlang.org/) gewechselt, um Fehler früher zu entdecken (und Mädchen haben wir viele gefunden) und nutzen jetzt [zustand](https://github.com/pmndrs/zustand) für das Statemanagement, in der Hoffnung, dass das Race-Condition-Bugs im Code behebt. Das heißt auch, dass wir die alten gespeicherten Anfragen in eine neue Speicherart überführt haben. Wir hoffen, dass alles gut gegangen ist, aber falls Du irgendwelche Probleme bemerkt hast und/oder irgendwelche Anfragen vermisst, [schreib uns eine E-Mail](mailto:dev@datenanfragen.de). Außerdem haben wir den Code ordentlich aufgeräumt und seine alten und dysfunktionalen Ecken entfernt. Und wir veröffentlichen jetzt [ein Package unserer Generators](https://www.npmjs.com/package/@datenanfragen/components), das Du in Deinen eigenen Code einbauen kannst, z.&nbsp;B. wenn du es in Deine Webseite einbetten willst.
```

## Hilfe gesucht: Fehlermeldungen, Unternehmensvorschläge und Übersetzungen

> Jetzt da wir die neuen Features veröffentlicht haben, brauchen wir eure Hilfe: Wir haben versucht alles intensiv zu testen, aber manchmal schlüpft eben doch ein Fehler druch. Falls Du also einen findest, schick uns gerne eine Fehlermeldung, entweder über unsere Fehlermeldungsfenster oder an [dev@datenanfragen.de](mailto:dev@datenanfragen.de). Außerdem haben sich viele Texte verändert oder wurden neu in unsere Übersetzungen hinzugefügt. Wenn Du mitmachen willst, kannst Du sie [via Weblate](https://hosted.weblate.org/projects/datenanfragen-de/website/) in andere Sprachen übersetzen, das wäre wahnsinnig toll! Schlißelich können wir nur in den Länder Unternehmen vorschlagen, für die wir auch Unternehmenspacks haben. Für Deustchland haben wir einen guten Überblick, aber falls Du denkst, dass Du für ein anderes Land gut weißt, welche Unternehmen regelmäßig die Daten von vielen Menschen in diesem Land verarbeiten, füg gerne ein Unternehmenspack [in unsere Datenbank](https://github.com/datenanfragen/data#company-packs) hinzu (Bitte gibt dabei ausführliche Quellen an).

Reviewer: ```suggestion
Jetzt, da wir die neuen Features veröffentlicht haben, brauchen wir eure Hilfe: Wir haben versucht alles intensiv zu testen, aber manchmal schlüpft eben doch ein Fehler durch. Falls Du also einen findest, schick uns gerne eine Fehlermeldung, entweder über unsere Fehlermeldungsfenster oder an [dev@datenanfragen.de](mailto:dev@datenanfragen.de). Außerdem haben sich viele Texte verändert oder wurden neu in unsere Übersetzungen hinzugefügt. Wenn Du mitmachen willst, kannst Du sie [via Weblate](https://hosted.weblate.org/projects/datenanfragen-de/website/) in andere Sprachen übersetzen, das wäre unfassbar toll! Schließlich können wir nur in den Länder Unternehmen vorschlagen, für die wir auch Unternehmenspacks haben. Für Deutschland haben wir einen guten Überblick, aber falls Du denkst, dass Du für ein anderes Land gut weißt, welche Unternehmen regelmäßig die Daten von vielen Menschen in diesem Land verarbeiten, füg gerne ein Unternehmenspack [in unsere Datenbank](https://github.com/datenanfragen/data#company-packs) hinzu (bitte gibt dabei ausführliche Quellen an).
```
